src/main/java/org/truffleruby/platform/Platform.java

```diff
@@ -287,18 +287,26 @@ private static boolean startsWithIgnoreCase(String s1, String s2) {
     }
 
     public static String getOSName() {
-        if (jnr.posix.util.Platform.IS_WINDOWS) {
+        // inlined jnr.posix.util.Platform.IS_WINDOWS
+        boolean isWindows = System.getProperty("os.name").toLowerCase().contains("windows");
+        if (isWindows) {
             return RUBY_WIN32;
         }
 
-        String OSName = jnr.posix.util.Platform.getOSName();
-        String theOSName = RUBY_OS_NAMES.get(OSName);
+        // inlined jnr.posix.util.Platform.getOSName()
+        String osNameProperty = System.getProperty("os.name");
+        final Map<String, String> jnrOsNames = new HashMap<String, String>();
+        jnrOsNames.put("Mac OS X", "darwin");
+        jnrOsNames.put("Darwin", "darwin");
+        jnrOsNames.put("Linux", "linux");
 
-        return theOSName == null ? OSName : theOSName;
+        String osName = jnrOsNames.getOrDefault(osNameProperty, osNameProperty);
+        return RUBY_OS_NAMES.getOrDefault(osName, osName);
     }
```

Comment:
We need to better preserve the licenses of `jar-posix` here. Maybe put the methods you're using into a separate file and include the license text there?